feature_details.md

```diff
@@ -0,0 +1,413 @@
+
+```
+// The list of elements is keyed on url. When a user navigates to
+// a url, the corresponding entry in this dictionary is used to
+// decide which elements should be cached.
+document.documentTransition.setSameOriginTransitionElements(
+  {
+   “foo.com/a.html” : [{element : document.html}],
+  });
+```
+
+Now we need to add some code on the new Document. When the new Document loads, script registers a listener for a new event type called `handleTransition`. The event is dispatched if a transition was initiated by the previous Document. The exact Document lifecycle stage before which this event should be registered is TBD.
+
+```
+document.documentTransition.addEventListener(
+    “handleTransition”, e => {
+        e.startTransition(provideTransitionFrom(e.previousUrl, e.previousElements));
+    });
+
+async function provideTransitionFrom(previousUrl, previousElements) {
+    // Wait for the image on the new page to load before
+    // starting the transition.
+    await waitForImageLoad();
+    
+    // Old content slides with a fade out from center to left.
+    let oldRootAnimation =
+      {previousElement : previousElements[0],
+       transitionType : "slide-fade-left"};
+    
+    // New content is drawn underneath, revealed as the old
+    // content slides.
+    let newRootAnimation =
+      {newElement : document.html,
+       transitionType : "none"};
+
+    return [oldRootAnimation, newRootAnimation];
+}
+```
+
+Adding an event listener for `handleTransition` will result in execution of following steps :
+
+* The browser keeps displaying content for the previous Document when `handleTransition` is dispatched. This does not require the previous Document to be active since the browser can display a cached pixel copy.
+
+* The `handleTransition` event has a field `previousURL` : the URL for the previous Document. And `previousElements` : the list of placeholders for elements passed to `setSameOriginTransitionElements`, clarified in [API Proposal](#api-proposal).
+
+* The `handleTransition` event has an API `startTransition` which takes the list of elements to animate and the type of animation. In the example above, `provideTransitionFrom` returns the root elements for the previous and new Document. Also note that the animation done is based on `transitionType` specified with each element. This is a new enum with a pre-defined list of animation patterns.
+
+* The `startTransition` API takes a promise to allow script to asynchronously prepare the new Document for first paint. The browser continues to display old content until the promise passed to this API resolves.
```

Comment:
The last two bullet points seem to contradict each other: does `startTransition` take a list of elements to animate or does it take a promise?

Reply:
Fixed the wording. Thanks!